docs: add AI Gateway rebranding migration guide (#26854)
coder/coder: docs/ai-coder/ai-gateway/rebranding-migration.md

# Rebranding Migration

> [!NOTE]
> AI Gateway requires the [AI Governance Add-On](../ai-governance.md).
> As of Coder v2.32, deployments without the add-on will not be able to
> access AI Gateway.

AI Bridge has been renamed to **AI Gateway**. This is a cosmetic rebrand to make
the feature easier to understand. It changes user-visible names, configuration
options, the canonical HTTP API path, and the Prometheus metric names.

> **This release does not break existing deployments.** Old names keep working as

Reviewer: nit: should we add this with a note?

```suggestion
> [!NOTE]
> This release does not break existing deployments. Old names keep working as
```

deprecated aliases, there are no database changes, and no configuration changes
are required to upgrade.

> The deprecated aliases may be removed in a future release, so we recommend

Reviewer: Did we already agree on a release? 🤔 If yes, it would be nice to add an estimated release for removal, so customers can plan accordinly

Author: I've raised it as a discussion point for today's Weekly call; I'm kinda in favour of keeping these around indefinitely tbh.

migrating as soon as possible.

> We will only use the new name for future settings, with no aliases.

Reviewer: nit: I feel this is important for operators, maybe using a Note (or Warning)? 🤔

Reviewer: Also, should we be specific and say that this affects all the different areas (except Database)?

| `CODER_AIBRIDGE_PROXY_UPSTREAM_CA`           | `CODER_AI_GATEWAY_PROXY_UPSTREAM_CA`           |                                   |
| `CODER_AIBRIDGE_PROXY_ALLOWED_PRIVATE_CIDRS` | `CODER_AI_GATEWAY_PROXY_ALLOWED_PRIVATE_CIDRS` |                                   |
| `CODER_AIBRIDGE_PROXY_DUMP_DIR`              | `CODER_AI_GATEWAY_PROXY_DUMP_DIR`              |                                   |
| `CODER_AIBRIDGE_PROXY_DOMAIN_ALLOWLIST`      | `CODER_AI_GATEWAY_PROXY_DOMAIN_ALLOWLIST`      | Already deprecated; has no effect |

CLI flags follow the same mapping with the `--aibridge-*` to `--ai-gateway-*`
prefix change.

### Provider configuration env vars

Providers are configured with indexed environment variables of the form
`CODER_AI_GATEWAY_PROVIDER_<N>_<KEY>` (for example
`CODER_AI_GATEWAY_PROVIDER_0_TYPE`, `CODER_AI_GATEWAY_PROVIDER_0_NAME`,
`CODER_AI_GATEWAY_PROVIDER_0_KEY`, `CODER_AI_GATEWAY_PROVIDER_0_BASE_URL`). The
old `CODER_AIBRIDGE_PROVIDER_<N>_<KEY>` prefix is accepted as a deprecated alias.

Unlike the scalar settings above, you **cannot mix the two prefixes**. Setting
both `CODER_AIBRIDGE_PROVIDER_*` and `CODER_AI_GATEWAY_PROVIDER_*` variables in

> the same deployment is an error. Move every provider variable onto the new

Reviewer: nit: should we show the error that operators get in this case?

`CODER_AI_GATEWAY_PROVIDER_*` prefix together (for example
`CODER_AIBRIDGE_PROVIDER_0_TYPE` becomes `CODER_AI_GATEWAY_PROVIDER_0_TYPE`).

## HTTP API

The canonical API path is now `/api/v2/ai-gateway` (and
`/api/v2/ai-gateway/proxy`). The legacy `/api/v2/aibridge` and
`/api/v2/aibridge/proxy` routes are retained for backward compatibility and
continue to serve the same handlers.

If you have external integrations calling the API directly, update them to the

> new path at your convenience. No immediate action is required.

Reviewer: We should also mention that AI client configuration should be updated to use the new base URL.

## Metrics

The metric prefixes have been renamed:

| Deprecated prefix        | New prefix                 |
|--------------------------|----------------------------|
| `coder_aibridged_*`      | `coder_ai_gateway_*`       |
| `coder_aibridgeproxyd_*` | `coder_ai_gateway_proxy_*` |

**Both the old and new metric names are emitted simultaneously today.** Every
series is exported under both prefixes from the same underlying collector, so
existing dashboards, alerts, and recording rules keep working immediately after
upgrade with no changes.

To avoid a gap in your observability when the old metric names are removed:

1. Update Grafana dashboards, Prometheus alerting rules, and recording rules to
   reference the new `coder_ai_gateway_*` and `coder_ai_gateway_proxy_*` names.
2. Verify the new series are present in your monitoring stack (they are emitted
   as of this release).

> ### Optional: dropping the old names

Reviewer: Nice ⭐

# Rebranding Migration

> [!NOTE]
> AI Gateway requires the [AI Governance Add-On](../ai-governance.md).
> As of Coder v2.32, deployments without the add-on will not be able to
> access AI Gateway.

AI Bridge has been renamed to **AI Gateway**. This is a cosmetic rebrand to make
the feature easier to understand. It changes user-visible names, configuration
options, the canonical HTTP API path, and the Prometheus metric names.

> [!NOTE]

> > This release does not break existing deployments. Old names keep working as

Reviewer 2: ```suggestion
> This release does not break existing deployments. Previous names keep working as
```

> deprecated aliases, there are no database changes, and no configuration
> changes are required to upgrade.

> The old `aibridge` names are retained for backward compatibility. There is no

Reviewer 2: ```suggestion
The previous `aibridge` names are retained for backward compatibility. There is no
```

> planned removal date, but we recommend adopting the new `ai_gateway` names so

Reviewer 2: ```suggestion
planned removal date, but you should adopt the new `ai_gateway` names as soon as possible, so
```

Nit: I avoid using any first person plural pronouns per Microsoft and Google's style guides

your configuration matches the current documentation.

> [!IMPORTANT]
> New settings added in every area except the database (configuration options,
> environment variables, CLI flags, and API paths) will use only the new
> `ai_gateway` name, with no `aibridge` alias.

## At a glance

> | Area                  | Old name                                       | New (canonical) name                              | Old name still works?              |

Reviewer 2: ```suggestion
| Area                  | Previous name                                  | New (canonical) name                              | Previous name still works?         |
```

This _should_ be formatted so as not to break the markdown table formatter.

Author: https://coder.com/docs/@dk%2Fgateway-migration-guide/ai-coder/ai-gateway/rebranding-migration#at-a-glance looks fine to me?

The `make lint/markdown` linter also doesn't auto-fix, so I think we're good?

|-----------------------|------------------------------------------------|---------------------------------------------------|------------------------------------|
| Environment variables | `CODER_AIBRIDGE_*`                             | `CODER_AI_GATEWAY_*`                              | Yes (deprecated alias)             |
| CLI flags             | `--aibridge-*`                                 | `--ai-gateway-*`                                  | Yes (deprecated alias)             |
| YAML config group     | `aibridge:` / `aibridgeproxy:`                 | `ai_gateway:` / `ai_gateway_proxy:`               | Yes (deprecated alias)             |
| HTTP API              | `/api/v2/aibridge`                             | `/api/v2/ai-gateway`                              | Yes (legacy route retained)        |
| Prometheus metrics    | `coder_aibridged_*` / `coder_aibridgeproxyd_*` | `coder_ai_gateway_*` / `coder_ai_gateway_proxy_*` | Yes (both emitted, old deprecated) |
| Database              | (no change)                                    | (no change)                                       | n/a                                |

## What did not change

> - **No database changes.** Table and column names (for example

Reviewer 2: ```suggestion
- **No database changes.** Table and column names (for example,
```

  `aibridge_interceptions`) are unchanged. No migration runs and no data is
  rewritten on upgrade.
- **No behavioral changes.** This is a naming change only. Values, defaults, and
  semantics of every option are identical.
- **Internal/library references.** Some internal package names, log fields, and
  library identifiers still use the `aibridge` name. These are not part of the
  supported configuration surface and do not affect operators.

## Configuration (env vars, flags, YAML)

> The new names are the canonical options; the old `aibridge` names still set the

Reviewer 2: ```suggestion
The new names are the canonical options; the previous `aibridge` names still set the
```

same values as hidden, deprecated aliases.

> If both an old and a new name are set for the same setting, set only one (prefer

Reviewer 2: ```suggestion
If both a previous name and a new name are set for the same setting, set only one (prefer
```

| `CODER_AIBRIDGE_ALLOW_BYOK`                        | `CODER_AI_GATEWAY_ALLOW_BYOK`                        |                                                                |
| `CODER_AIBRIDGE_CIRCUIT_BREAKER_ENABLED`           | `CODER_AI_GATEWAY_CIRCUIT_BREAKER_ENABLED`           |                                                                |
| `CODER_AIBRIDGE_CIRCUIT_BREAKER_FAILURE_THRESHOLD` | `CODER_AI_GATEWAY_CIRCUIT_BREAKER_FAILURE_THRESHOLD` |                                                                |
| `CODER_AIBRIDGE_CIRCUIT_BREAKER_INTERVAL`          | `CODER_AI_GATEWAY_CIRCUIT_BREAKER_INTERVAL`          |                                                                |
| `CODER_AIBRIDGE_CIRCUIT_BREAKER_TIMEOUT`           | `CODER_AI_GATEWAY_CIRCUIT_BREAKER_TIMEOUT`           |                                                                |
| `CODER_AIBRIDGE_CIRCUIT_BREAKER_MAX_REQUESTS`      | `CODER_AI_GATEWAY_CIRCUIT_BREAKER_MAX_REQUESTS`      |                                                                |

> | `CODER_AIBRIDGE_PROVIDER_<N>_<KEY>`                | `CODER_AI_GATEWAY_PROVIDER_<N>_<KEY>`                | Cannot be mixed; see [below](#provider-configuration-env-vars) |

Reviewer 2: ```suggestion
| `CODER_AIBRIDGE_PROVIDER_<N>_<KEY>`                | `CODER_AI_GATEWAY_PROVIDER_<N>_<KEY>`                | Cannot be mixed; [more info](#provider-configuration-env-vars) |
```

| `CODER_AIBRIDGE_PROXY_CERT_FILE`             | `CODER_AI_GATEWAY_PROXY_CERT_FILE`             |                                   |
| `CODER_AIBRIDGE_PROXY_KEY_FILE`              | `CODER_AI_GATEWAY_PROXY_KEY_FILE`              |                                   |
| `CODER_AIBRIDGE_PROXY_UPSTREAM`              | `CODER_AI_GATEWAY_PROXY_UPSTREAM`              |                                   |
| `CODER_AIBRIDGE_PROXY_UPSTREAM_CA`           | `CODER_AI_GATEWAY_PROXY_UPSTREAM_CA`           |                                   |
| `CODER_AIBRIDGE_PROXY_ALLOWED_PRIVATE_CIDRS` | `CODER_AI_GATEWAY_PROXY_ALLOWED_PRIVATE_CIDRS` |                                   |
| `CODER_AIBRIDGE_PROXY_DUMP_DIR`              | `CODER_AI_GATEWAY_PROXY_DUMP_DIR`              |                                   |
| `CODER_AIBRIDGE_PROXY_DOMAIN_ALLOWLIST`      | `CODER_AI_GATEWAY_PROXY_DOMAIN_ALLOWLIST`      | Already deprecated; has no effect |

CLI flags follow the same mapping with the `--aibridge-*` to `--ai-gateway-*`
prefix change.

### Provider configuration env vars

Providers are configured with indexed environment variables of the form

> `CODER_AI_GATEWAY_PROVIDER_<N>_<KEY>` (for example

Reviewer 2: ```suggestion
`CODER_AI_GATEWAY_PROVIDER_<N>_<KEY>` (for example,
```

`CODER_AI_GATEWAY_PROVIDER_0_TYPE`, `CODER_AI_GATEWAY_PROVIDER_0_NAME`,
`CODER_AI_GATEWAY_PROVIDER_0_KEY`, `CODER_AI_GATEWAY_PROVIDER_0_BASE_URL`). The
old `CODER_AIBRIDGE_PROVIDER_<N>_<KEY>` prefix is accepted as a deprecated alias.

Unlike the scalar settings above, you **cannot mix the two prefixes**. Setting
both `CODER_AIBRIDGE_PROVIDER_*` and `CODER_AI_GATEWAY_PROVIDER_*` variables in
the same deployment causes startup to fail with:

```text
cannot mix CODER_AIBRIDGE_PROVIDER_* and CODER_AI_GATEWAY_PROVIDER_* environment variables, please consolidate onto CODER_AI_GATEWAY_PROVIDER_*
```

Move every provider variable onto the new `CODER_AI_GATEWAY_PROVIDER_*` prefix

> together (for example `CODER_AIBRIDGE_PROVIDER_0_TYPE` becomes

Reviewer 2: ```suggestion
together (for example, `CODER_AIBRIDGE_PROVIDER_0_TYPE` becomes
```
